### ISSUE TITLE
Property testing don't accept suspended lambdas

### ISSUE THREAD

**Issue by LeoColman** [kotest/kotest#717]
In the same fashion of https://github.com/kotlintest/kotlintest/issues/707, the following code won't work:

```
"Foo" - {
     forAll<Int> { first: Int -> 
        "Bar" { println(first) } 
    } 
}

```

**Comment by LeoColman** [comment]
@sksamuel 
One approach I was trying for this was inlining everything.
However, this leads to a forever-running compilation process, as you'll see for the commit.

We'll have to look for a alternative on this. Perhaps doing the suspend-version again. But this is starting to become a bigger problem

**Comment by sksamuel** [comment]
I think we should try to address this soon, maybe we can take a fix for things like this as 3.4 release while we continue to work on 4.0 in the background.

**Comment by LeoColman** [comment]
This is nearly done
